clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -2923,14 +2923,20 @@ static void handleWorkGroupSize(Sema &S, Decl *D, const ParsedAttr &AL) {
     return;
 
   uint32_t WGSize[3];
```

Comment:
Note: Every comparison with this gives me a -Wsign-compare warning.  XDim, YDim, and ZDim are all 'int' in type.  This should be so as well likely.

Reply:
The problem is TableGen has no unsigned type.

Reply:
Okey, this problem I solved, it seems.